uaclient/tests/test_sso.py

```diff
@@ -122,6 +140,37 @@ def test_2fa_errors_dont_raise_and_request_extra_input(
 
         assert mock.call('Second-factor auth: ') in m_input.call_args_list
 
+    @pytest.mark.parametrize(
+        'error_message,reprompt',
+        (('The provided 2-factor key is not recognised.', True),
+         ('Invalid email or password', False)))
+    @mock.patch('uaclient.sso.getpass')
+    @mock.patch('builtins.input')
+    @mock.patch('uaclient.sso.UbuntuSSOClient')
+    def test_2fa_credential_errors_reprompt_for_second_factor_auth_input(
+            self, m_sso_client, m_input, _m_getpass, error_message, reprompt):
```

Comment:
`reprompt` is never used.

Reply:
I forgot to use it; instead performing a string search in the unit test. I've changed the test to make use of reprompt now.